packages/agent-utils/src/memory/LocalDocumentStore.ts

```diff
@@ -0,0 +1,35 @@
+import { Workspace } from "../sys";
+import path from "path-browserify";
+import { v4 as uuid } from "uuid";
+import { DocumentMetadata, LocalDocument } from "./LocalDocument";
+
+export class LocalDocumentStore {
+  constructor(
+    private workspace: Workspace,
+    private uri: string
+  ) {}
+
+  add(data: { text: string, vector: number[], metadata?: DocumentMetadata }): LocalDocument {
+    const id = uuid()
+
+    const document = new LocalDocument(id, {
```

Comment:
Didn't go too much into it, but it seems like more logic from the LocalDocument should be in the LocalDocumentStore